File: problems/2383-minimum-hours-of-training-to-win-a-competition/solution.java
class Solution {
    public int minNumberOfHours(int initialEnergy, int initialExperience, int[] energy, int[] experience) {
        int energyNeeded = 0;
        int experienceNeeded = 0;

        int sumEnergy = 0;
        for (int e : energy) {
            sumEnergy += e;
        }

        if (initialEnergy <= sumEnergy) {
            energyNeeded = sumEnergy - initialEnergy + 1;
        }

        int currentExperience = initialExperience;
        for (int exp : experience) {
            if (currentExperience <= exp) {
                experienceNeeded += (exp - currentExperience + 1);
                currentExperience = exp + 1;
            }
            currentExperience += exp;
        }

        return energyNeeded + experienceNeeded;
    }
}

File: problems/2383-minimum-hours-of-training-to-win-a-competition/solution.py
class Solution:
    def minNumberOfHours(self, initialEnergy: int, initialExperience: int, energy: List[int], experience: List[int]) -> int:
        energy_needed = 0
        for e in energy:
            if initialEnergy <= e:
                energy_needed += (e - initialEnergy + 1)
                initialEnergy = e + 1
            initialEnergy -= e
        
        experience_needed = 0
        curr_experience = initialExperience
        for exp in experience:
            if curr_experience <= exp:
                experience_needed += (exp - curr_experience + 1)
                curr_experience = exp + 1
            curr_experience += exp
        
        return energy_needed + experience_needed